runtime/src/bank.rs

```diff
@@ -3135,8 +3135,8 @@ impl Bank {
         );
 
         inc_new_counter_debug!("bank-register_tick-registered", 1);
-        let mut w_blockhash_queue = self.blockhash_queue.write().unwrap();
         if self.is_block_boundary(self.tick_height.load(Relaxed) + 1) {
+            let mut w_blockhash_queue = self.blockhash_queue.write().unwrap();
```

Comment:
Yeah we only call it from one thread right now

Reply:
If it's not thread safe to move the lock, and only one thread is running this, then the only point in moving the write lock would only save the call to `self.is_block_boundary()`? Seems like it might not be worth the churn

Reply:
Since validators should typically be read-favored, I was worried about write starvation as well. Ticks could be delayed because we grab the read lock a lot for the blockhash queue. For example, nonced transactions hold the read lock while loading and deserializing nonce accounts. As long as there's an active read, the write has to wait and there's no sense in doing that for each of the 64 ticks per slot.

Reply:
We also hold the blockhash queue read lock the entire time while loading accounts for a tx

Reply:
makes sense, could be why `flush_cache()` is often one of the most expensive parts of the `poh_recorder` metric